app/models/certificate.server.ts

```diff
@@ -0,0 +1,64 @@
+import type { Certificate } from '@prisma/client';
+
+import { prisma } from '~/db.server';
+
+export type { Certificate } from '@prisma/client';
+
+export async function getCertificateByUsername(username: Certificate['username']) {
+  return prisma.certificate.findUnique({ where: { username } });
+}
+
+export async function createCertificate(
+  username: Certificate['username'],
+  subject: Certificate['subject'],
+  certificate: Certificate['certificate'],
+  orderUrl: Certificate['orderUrl'],
+  privateKey: Certificate['privateKey']
+) {
+  // Set expiration date 90 days from now
+  const validTo = new Date();
+  validTo.setDate(validTo.getDate() + 90);
+
+  return prisma.certificate.create({
+    data: {
+      username,
```

Comment:
Instead of specifying the username, I think you should connect the existing user using `connect`: https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-an-existing-record

I tested your create function, and got this error in the db studio:
```
Inconsistent query result: Field user is required to return data, got `null` instead.
  
Query:
{
  "modelName": "Certificate",
  "operation": "findMany",
  "args": {
    "take": 100,
    "skip": 0,
    "select": {
      "id": true,
      "username": true,
      "subject": true,
      "certificate": true,
      "orderUrl": true,
      "privateKey": true,
      "validFrom": true,
      "validTo": true,
      "user": true,
      "challenge": {
        "select": {
          "id": true
        }
      }
    }
  }
}
```

Reply:
```suggestion
      user: { connect: { username } }
```

I tested with this, and that resolves the error

Reply:
Thanks for showing me that. That approach can work, but I would need to make another API call to update `user`, so I think the current method is simpler.

As for the error, I didn't run into any error message.
Can you give me more details on this? What did you do to have the query executing? Where is the error displayed? 
I didn't run into any error messages.

Reply:
From the error, it looks like you were trying to do something with `findMany()` on `Certificate` model. It's not a function I made or used in this PR for 'Certificate'. It's just a default function from Prisma itself.

Reply:
The findMany() is used by the db studio to get all the certificates. Try creating a certificate, and then open db studio in the browser to see this error. I will update with a screenshot shortly

Reply:
Hmmm can't replicate the error anymore. Not sure why, but I guess there is no need to worry about this. Previously, it seemed like the certificate was not connecting to the user. I got the error in the browser

Reply:
@cychu42  Could you please confirm that this does not happen for you:
- Use a query file to create a new user, and then a new certificate for the user
- Run `npm run db:studio` and check if the error appears in your browser

Reply:
I received no error.
Maybe there was mismatch in your previous input for username, or an user got incorrectly removed? Just guesses.

Reply:
Yeah, I tried removing a user but you can't remove a user without removing the certificate. I am thinking maybe my local environment was messed up